caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java

```diff
@@ -88,20 +80,21 @@ public URL toURL(URI uri) {
     private URL createURLFromPath(URI uri) {
         URL newUrl = null;
         String[] path = uri.getSchemeSpecificPart().split("/");
-        if (path.length != 2) {
-            throw new IllegalArgumentException("Malformed URI. Expected 2 path components, found " + path.length);
+        // data uri has 3 parts, preview has 2
+
+        if (path.length < 2 || path.length > 3) {
+            throw new IllegalArgumentException("Malformed URI. Expected 2 or 3 path components, found " + path.length);
         }
 
-        String requestType = path[0];
-        String fileName = path[1];
         String sb = "";
-
-        if (requestType.equals(FILE_URI)) {
-            // Returns a quick search-style URL for SMOKA Search page
-            sb = FILE_BASE_URL + "?" + FILE_URL_QUERY + fileName;
-        } else if (requestType.equals(PREVIEW_URI)) {
+        String requestType = path[0];
+        if (path.length == 2 && requestType.equals(PREVIEW_URI)) {
             // Returns a web page reference
-            sb = PREVIEW_BASE_URL + "?" + PREVIEW_URL_QUERY + fileName;
+            sb = PREVIEW_BASE_URL + "?" + PREVIEW_URL_QUERY + path[1];
+        } else if (path.length == 3 && requestType.equals(RAW_DATA_URI)) {
+            // expected URI input is subaru:data/YYYY-MM-dd/<FRAMEID>
+            // expected URL output is http://www.canfar.net/maq/subaru?frameinfo=YYYY-MM-dd%20FRAMEID
+            sb = BASE_DATA_URL +  NetUtil.encode(path[1] + "/" + path[2]);
         } else {
             throw new IllegalArgumentException("Invalid URI. Expected 'file' or 'preview' and got " + requestType);
```

Comment:
The mention of 'file' in the error message doesn't match declared constant values; that will be misleading if (when) it happens